### PR TITLE
4632-pills-badges

### PR DESCRIPTION
**Pull request by sarrailhadaptive**

### CODE REVIEW

File: components/src/pill/Pill.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+    if (isDisabled && onLightBg) {
+      style += "bg-primary-light pointer-events-none opacity-60"
+    }
+    if (isDisabled && !onLightBg) {
+      style += "bg-transparent pointer-events-none opacity-60"
+    }
+  } else if (purpose === "ticker") {
+    style = "w-16 bg-transparent rounded-xl px-2 py-1 text-sm font-medium text-primary-main border-2 border-primary-main hover:border-primary-light hover:text-primary-light"
+  }
```
Comment (adamshcao) [inline components/src/pill/Pill.tsx:60]: Would a switch statement shorten this? Also should look into abstracting repeated groups of tailwind classes into single CSS classes like discussed in standup. https://tailwindcss.com/docs/reusing-styles

File: components/src/App.tsx [hunk 2]
```diff
@@ -27,6 +28,7 @@ function App() {
       <p className="read-the-docs">
         Click on the Vite and React logos to learn more
       </p>
+      <Pill purpose="status" onLightBg={false} isDisabled={false} type="success" text="success" />
```
Comment (adamshcao) [inline components/src/App.tsx:31]: Remove before merge

File: components/src/App.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { useState } from "react"
 import reactLogo from "./assets/react.svg"
 import "./App.css"
+import { Pill } from "./pill/Pill"
```
Comment (adamshcao) [inline components/src/App.tsx:4]: Remove before merge

File: components/src/pill/Pill.tsx (hunk 1 above)
Comment (sarrailhadaptive) [reply components/src/pill/Pill.tsx]: Tailwind classes have been fixed and the code shortened, I didn't go with the "switch" option to replace "if" statements for readability and simplicity.

File: components/src/App.tsx (hunk 2 above)
Comment (sarrailhadaptive) [reply components/src/App.tsx]: Fixed

File: components/src/App.tsx (hunk 3 above)
Comment (sarrailhadaptive) [reply components/src/App.tsx]: Fixed

File: components/test/pill.test.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+function toJson(component: renderer.ReactTestRenderer) {
+  const result = component.toJSON()
+  expect(result).toBeDefined()
+  expect(result).not.toBeInstanceOf(Array)
+  return result as renderer.ReactTestRendererJSON
+}
+
+test("Status Success Pill DarkMode", () => {
+  const component = renderer.create(<Pill purpose="status" onLightBg={false} isDisabled={false} type="success" text="success" />)
```
Comment (algreasley) [inline components/test/pill.test.tsx:14]: minor: could probably extract this to utility function .. 
I do not always suggest DRY techniques for tests as it is nice to see what each test is doing, especially if setup data is involved, but in this case I would suggest it is useful as you can more easily compare tests and see what is different in the inputs, without having to process the repeated bits like `tojson` etc.
(as I say, a minor thing, more opinion than absolute best practice 😄 )

File: components/tailwind.config.cjs [hunk 5]
```diff
@@ -80,4 +80,4 @@ module.exports = {
     },
   },
   plugins: [],
-}
+}
```
Comment (algreasley) [inline components/tailwind.config.cjs:83]: revert this file, probably

File: docs/src/stories/Pill.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import React from "react"
+import { ComponentStory, ComponentMeta } from "@storybook/react"
```
Comment (algreasley) [inline docs/src/stories/Pill.stories.tsx:2]: minor: deprecated types .. use
```
import { Meta, StoryFn } from "@storybook/react"
```

File: docs/src/stories/Pill.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+import { Pill } from "components"
+import "components/dist/style.css"
+
+export default {
+  title: "Example/Pill",
```
Comment (algreasley) [inline docs/src/stories/Pill.stories.tsx:8]: -> "Components/Pill"

File: docs/src/stories/Pill.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  component: Pill,
+} as ComponentMeta<typeof Pill>
+
+const Template: ComponentStory<typeof Pill> = (args) => <Pill {...args} />
+
+export const Success = Template.bind({})
```
Comment (algreasley) [inline docs/src/stories/Pill.stories.tsx:14]: I think switching between 'variants' should be via SB Controls (see Checkbox as best example of this to date)

**Review by algreasley — COMMENTED** [review]
looks good .. I think just needs to be made consistent with the existing components (using Controls) and I think it is not reacting well to light/dark right now

File: components/test/pill.test.tsx (hunk 4 above)
Comment (sarrailhadaptive) [reply components/test/pill.test.tsx:14]: Makes sense, I'm gonna re factor

File: components/tailwind.config.cjs (hunk 5 above)
Comment (algreasley) [reply components/tailwind.config.cjs:92]: this is still coming up as a diff ...

Comment (sarrailhadaptive) [reply components/tailwind.config.cjs:92]: Still happening, it seems like it deletes the curly bracket and adds it again, but it makes no sense that it only does that. Formatting on save works but when I run yarn format it doesn't target this file. I'm gonna check why this is happening and push again.

File: components/src/pill/Pill.tsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
+import React from "react"
+import "./pill.css"
+
+interface PillType {
+  // Is the pill a Primary or Status Pill
+  purpose: string
+  // Is the type of the pill a Success, Warning, Buy or Sell Pill -- Only for Status Pills
+  onLightBg: boolean
```
Comment (algreasley) [inline components/src/pill/Pill.tsx:8]: should not need, or be using `onLightBg` now that we have the light/dark toggle, right?

File: components/tailwind.config.cjs (hunk 5 above)
Comment (sarrailhadaptive) [reply components/tailwind.config.cjs]: Resolved.

File: components/src/pill/Pill.tsx (hunk 9 above)
Comment (sarrailhadaptive) [reply components/src/pill/Pill.tsx:8]: Resolved.

**Review by algreasley — APPROVED** [review]
great stuff .. thanks for adding in the formatting enhancements!

**Pull request merged**